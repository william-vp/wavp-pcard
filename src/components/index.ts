import {ProductCard as ProductCardComponent} from "./ProductCard";
import {ProductTitle} from "./ProductTitle";
import {ProductImage} from "./ProductImage";
import {ProductButtons} from "./ProductButtons";
import {ProductCardMainProps} from "../interfaces/interfaces";

export {ProductTitle} from "./ProductTitle";
export {ProductImage} from "./ProductImage";
export {ProductButtons} from "./ProductButtons";

export const ProductCard: ProductCardMainProps = Object.assign(ProductCardComponent, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons
});
export default ProductCard;