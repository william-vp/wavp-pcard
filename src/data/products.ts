import {Product} from "../interfaces/interfaces";
import coffeMug from "../coffee-mug.png";
import coffeMug2 from "../coffee-mug2.png";

const product = {
    id: '1',
    title: 'Coffe Mug',
    img: coffeMug
}

const product2 = {
    id: '2',
    title: 'Coffe Mug 2',
    img: coffeMug2
}
export const products: Product[] = [product, product2]