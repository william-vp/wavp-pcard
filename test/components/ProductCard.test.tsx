import React from 'react';
import {ProductCard} from "../../src";
import render from "react-test-renderer";
import {product1, product2} from "../data/products";

const {act}= render

describe('ProductCard', () => {
    test('mostrar componente', () => {
        const wrapper = render.create(
            <ProductCard product={product1}>
                { () => (
                    <h1>Product 1</h1>
                )}
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
    
    test('contador', () => {
        const wrapper = render.create(
            <ProductCard product={product2} >
                { ({count, increment}) => (
                    <>
                        <span>{count}</span>
                        <button onClick={() => increment(1)}>+1</button>
                    </>
                )}
            </ProductCard>
        );
        let tree= wrapper.toJSON();
        console.log((tree as any).children[1]);
        expect(tree).toMatchSnapshot();
        
        act(() => {
            (tree as any).children[1].props.onClick();
        });
        
        
        tree= wrapper.toJSON();
        console.log((tree as any).children[0]);
        expect((tree as any).children[0].children[0]).toMatchSnapshot();
        
    });
    
    
});
