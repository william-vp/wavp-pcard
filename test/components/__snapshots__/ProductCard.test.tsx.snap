// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard contador 1`] = `
<div
  className="productCard "
>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;

exports[`ProductCard contador 2`] = `"1"`;

exports[`ProductCard mostrar componente 1`] = `
<div
  className="productCard "
>
  <h1>
    Product 1
  </h1>
</div>
`;
